### PR TITLE
Update notes_controller.rb #3669

### LINKED ISSUE #3669: Update notes_controller.rb
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [app/controllers/notes_controller.rb](https://github.com/publiclab/plots2/blob/master/app/controllers/notes_controller.rb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://publiclab.github.io/community-toolbox/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -131,6 +131,7 @@ def create
           @node.add_tag('event:rsvp', current_user)
           @node.add_tag('date:' + params[:date], current_user) if params[:date]
         end
+        @node.add_tag('first-time-poster') if current_user.first_time_poster
         if params[:draft] != "true"
           if current_user.first_time_poster
             flash[:first_time_post] = true
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

(This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).)

### PR DESCRIPTION
**Pull request by quinn-codes-synthesis**
Fixes #3669 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by quinn-codes-synthesis** [comment]
Just a note: I tried the `rake test` thing locally as instructed and got some errors. They didn't make any sense to me, though. I only added the one line as instructed in the issue #3669 . Anyway, it's ready for review! @publiclab/reviewers

**Comment by jywarren** [comment]
Oops, maybe I made a mistake in my issue... Looks like it wants 2
parameters?

File: app/controllers/notes_controller.rb [hunk 1]
```diff
@@ -131,6 +131,7 @@ def create
           @node.add_tag('event:rsvp', current_user)
           @node.add_tag('date:' + params[:date], current_user) if params[:date]
         end
+        @node.add_tag('first-time-poster') if current_user.first_time_poster
```
Comment (jywarren) [inline app/controllers/notes_controller.rb]: Ah, oops! We need `, curren_user)` within the add_tag call. My bad!

**Comment by quinn-codes-synthesis** [comment]
`rake test` is still giving 2 failures and 1 error locally, but... the one line is fixed now. 

> ERROR["test_digest_emails", Minitest::Result, 15.200602795999998]
>  test_digest_emails#Minitest::Result (15.20s)
> Redis::CannotConnectError:         Redis::CannotConnectError: Error connecting to Redis on 127.0.0.1:6379 (Errno::ECONNREFUSED)
>             app/controllers/users_controller.rb:309:in `test_digest_email'
>             test/functional/users_controller_test.rb:258:in `block in <class:UsersControllerTest>'
> 
>  FAIL["test_search_question_at_/search/questions/question", Minitest::Result, 65.63406810400011]
>  test_search_question_at_/search/questions/question#Minitest::Result (65.63s)
>         Expected: 8
>           Actual: 9
>         test/functional/search_controller_test.rb:35:in `block in <class:SearchControllerTest>'
> 
>  FAIL["test_running_search_questions", Minitest::Result, 65.70335561600007]
>  test_running_search_questions#Minitest::Result (65.70s)
>         Expected: 2
>           Actual: 3
>         test/unit/api/search_service_test.rb:59:in `block in <class:SearchServiceTest>'

**Comment by jywarren** [comment]
Hmm, locally I believe the command is "rails test" now, but I admit I
haven't run tests locally for a while so I don't know if it makes a
difference.

https://github.com/publiclab/plots2/blob/master/doc/TESTING.md

**Comment by jywarren** [comment]
But tests passed here, so I think this is ready to go!

BTW thanks for being so helpful to others around the project. I'm traveling so not quite as able to write back to people as quick - I really appreciate it!

**Comment by jywarren** [comment]
I'd like to help you solve the errors, so I can merge this but there's no rush!

**Comment by jywarren** [comment]
Also it's late here now so I think I may circle back in the morning :-)

Thanks!

**Comment by quinn-codes-synthesis** [comment]
No worries, happy to help!

Since it passed all the tests on your end, go ahead and merge it. I was sharing the errors because I wasn't sure if they were something to be concerned about or not. (They might as well be hieroglyphics to me, ha.)

And yes, I think I saw you mention in the chat group that you were visiting China? That actually puts us on roughly the same schedule... I'm in EST but working the night shift for the time being.

Anyway, rest up, thanks for the guidance!

**Comment by gauravano** [comment]
Great work @quinn-codes-synthesis! 

Locally, digest_email_test requires redis and that's why test fails, we should open a new issue for that. Also, @jywarren do you think there is need of a follow-up issue needed which would remove the tag from node upon generation of second node by user? Thanks!

**Comment by gauravano** [comment]
@quinn-codes-synthesis just a suggestion  - Opening a PR on separate feature-branch can help you in keeping your `master` branch clean.

**Comment by jywarren** [comment]
Hi @gauravano - I think this one is OK because it'll be great to have a record of which posts were peoples' first posts, even later. However it'd be great to open a new issue for the Redis test thing. I think we could probably have a conditional around it like we do for full-text search tests?

Merging now, thanks, all! 🎉

**Pull request merged**